src/runtime/query/match/utils.ts

```diff
@@ -101,5 +101,7 @@ export const assertArray = (value: any, message = 'Expected an array') => {
 /**
  * Ensure result is an array
  */
-export const ensureArray = <T>(value: T) =>
-(Array.isArray(value) ? value : value ? [value] : []) as T extends Array<any> ? T : T[]
+export const ensureArray = <T>(value: T) => {
+  if (value === '') { return [''] } // Special case for empty string which is a valid value in an array
+  return (Array.isArray(value) ? value : value ? [value] : []) as T extends Array<any> ? T : T[]
```

Comment:
I think we can improve this by checking for `undefiend` and `null` values. With this it preserves empty string, `false` and `0`.

```suggestion
  return (Array.isArray(value) ? value : [null, undefined].includes(value) ? [] : [value]) as T extends Array<any> ? T : T[]
```

WDYT?